extensions/cache/runtime/src/main/java/io/quarkus/cache/runtime/caffeine/CacheValue.java

```diff
@@ -0,0 +1,43 @@
+package io.quarkus.cache.runtime.caffeine;
+
+import java.time.Duration;
+
+public class CacheValue<T> {
+
+    CacheValue(T data, Duration expiresIn) {
+        this.data = data;
+        this.expiresIn = expiresIn;
+    }
+
+    public static <T> CacheValueBuilder<T> builder() {
+        return new CacheValueBuilder<>();
+    }
+
+    public static class CacheValueBuilder<T> {
```

Comment:
I don't think that the builder is useful here..

Reply:
if we go this way yes for sure, I have another version of this PR where this is part of the API
`Uni<V> get(K key, Function<K, CacheValue<V>> valueLoader);`
which in my opinion allow more flexibility but also a breaking change.

If we decide not to use it in the API I will remove the builder and make it package private

Reply:
Ok, makes sense.